sdk/core/azure-core/src/main/java/com/azure/core/implementation/JsonPatchOperation.java

```diff
@@ -102,35 +104,116 @@ public boolean equals(Object obj) {
 
     @Override
     public String toString() {
-        return buildString(new StringBuilder()).toString();
+        return toJson(new StringBuilder()).toString();
     }
 
-    /**
-     * Appends this operation's JSON to the passed {@link StringBuilder}.
-     *
-     * @param builder The {@link StringBuilder} where this operation's JSON is being appended.
-     * @return The updated {@link StringBuilder}.
-     */
-    public StringBuilder buildString(StringBuilder builder) {
-        builder.append("{\"op\":\"")
+    @Override
+    public StringBuilder toJson(StringBuilder stringBuilder) {
+        stringBuilder.append("{\"op\":\"")
             .append(op.toString())
             .append("\"");
 
         if (from != null) {
-            builder.append(",\"from\":\"")
+            stringBuilder.append(",\"from\":\"")
                 .append(from)
                 .append("\"");
         }
 
-        builder.append(",\"path\":\"")
+        stringBuilder.append(",\"path\":\"")
             .append(path)
             .append("\"");
 
         if (value.isInitialized()) {
-            builder.append(",\"value\":")
+            stringBuilder.append(",\"value\":")
                 .append(value.getValue());
         }
 
-        return builder.append("}");
+        return stringBuilder.append("}");
+    }
+
+    @Override
+    public JsonWriter toJson(JsonWriter jsonWriter) {
+        // Write the start object and "op" property.
+        jsonWriter.writeStartObject().writeStringField("op", op.toString());
+
+        // Only write "from" property if it isn't null.
+        if (from != null) {
+            jsonWriter.writeStringField("from", from);
+        }
+
+        // Write the "path" property.
+        jsonWriter.writeStringField("path", path);
+
+        // Only write the "value" property if it exists.
+        if (value.isInitialized()) {
```

Comment:
do we need to support different serialization rules (e.g. on null values) at all?

Reply:
At least for JsonPatchOperation we need to differentiate between non-existent value and null value, in most other cases this shouldn't be needed.